uptrends/client.go

```diff
@@ -0,0 +1,68 @@
+package uptrends
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type clientor interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type requestCreator interface {
+	NewRequest(method, path string, body io.Reader) (*http.Request, error)
+}
+
+type httpReqCreator struct{}
+
+func (h *httpReqCreator) NewRequest(method, path string, body io.Reader) (*http.Request, error) {
+	return http.NewRequest(method, path, body)
+}
+
+// Client is an Uptrends.com API Client
+type Client struct {
+	baseURL url.URL
+	clientor
+	username string
+	password string
+	requestCreator
+}
+
+// New returns an Uptrends.com API Client
+//  username is the Uptrends.com login user
+//  password is the corresponding password for username
+func New(username string, password string) (*Client, error) {
+	u, err := url.Parse("https://api.uptrends.com")
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		*u,
+		&http.Client{
+			Timeout: time.Second * 10,
+		},
+		username,
+		password,
+		&httpReqCreator{},
+	}, nil
+}
+
+func (c *Client) newRequest(method, path string, body io.Reader) (*http.Request, error) {
+	url := c.baseURL
+	url.Path = strings.Join([]string{"", "v3", path}, "/")
```

Comment:
Should "v3" be a parameter for newRequest?

Reply:
Yeah, this was another thing I was wrestling with.
Do you suppose `v3` should be a part of the `newRequest(...)` call instead of the origination of the `Client` itself?
Is there any justification for having a `v3`-only client? I suppose I can just keep it in the `path` that's already being passed in with the `newRequest(...)`.

Definitely open to hearing how anyone might think through this.

Reply:
I think it's perfectly valid to have a `v3`-only client if that meets your use case. I think making that explicit in the README or documentation might be helpful if that is the case.